src/vivarium_census_prl_synth_pop/results_processing/generate_names.py

```diff
@@ -0,0 +1,63 @@
+import glob
+import os
+from pathlib import Path
+from typing import Tuple
+
+import pandas as pd
+from loguru import logger
+
+from vivarium_census_prl_synth_pop.components.synthetic_pii import NameGenerator
+
+COLUMNS_FOR_NAME_PROCESSING = [
+    "first_name_id",
+    "middle_name_id",
+    "last_name_id",
+    "date_of_birth",
+    "sex",
+    "race_ethnicity",
+]
+
+
+def load_data(raw_results_dir: Path):
+    data = pd.DataFrame()
+
+    observer_directories = [f.path for f in os.scandir(raw_results_dir) if f.is_dir()]
+    for obs_dir in observer_directories:
+        logger.info(f'Processing data for {obs_dir.split("/")[-1]}...')
+        results = glob.glob(os.path.join(obs_dir, "*.csv.bz2"))
+        for file in results:
+            tmp = pd.read_csv(file)
+            data = pd.concat([data, tmp])
+
+    # Drop unnecessary columns (Social Security Observer does not have sex  and race ethnicity)
+    data = data[COLUMNS_FOR_NAME_PROCESSING].drop_duplicates()
+
+    return data
+
+
+def fromat_data_for_name_generation(
+    data: pd.DataFrame,
+) -> Tuple:
+    # todo: update to handle with seed column once we are doing parallel runs
+    seed = 0
+
+    # Clean data
+    data["first_name_id"] = f"{seed}_" + data["first_name_id"].astype(str)
+    data["middle_name_id"] = f"{seed}_" + data["middle_name_id"].astype(str)
+    data["last_name_id"] = f"{seed}_" + data["last_name_id"].astype(str)
+    # Get year of birth - string column and not timestamp
```

Comment:
Was this the outcome of the difficulties we were having with saving to CSV format?

Reply:
No that was for changing the date of birth string in a separate ticket and the problem occurs when you read in the csv.  For example, March 1st, 2020 written as 03012020 gets interpreted as an integer when you do read_csv so the leading 0 is dropped.  Quoting option for write_csv seemed to not work with the date format changed to a string either.